pkg/volumes/openstack/volumes.go

```diff
@@ -324,9 +325,38 @@ func (stack *OpenstackVolumes) FindVolumes() ([]*volumes.Volume, error) {
 	return volumes, nil
 }
 
+func GetDevicePath(volumeID string) string {
+	// Build a list of candidate device paths
+	candidateDeviceNodes := []string{
+		// KVM
+		fmt.Sprintf("virtio-%s", volumeID[:20]),
+		// KVM virtio-scsi
+		fmt.Sprintf("scsi-0QEMU_QEMU_HARDDISK_%s", volumeID[:20]),
+		// ESXi
+		fmt.Sprintf("wwn-0x%s", strings.Replace(volumeID, "-", "", -1)),
+	}
+
+	files, _ := ioutil.ReadDir(volumes.PathFor("/dev/disk/by-id/"))
```

Comment:
Nit: we probably should at least log the error if there is one here, we may want to just return immediately (I'd expect files to be empty)